src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SORT {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function sort(goods: string[], sortField: SORT, isReversed: boolean): string[] {
+  const sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SORT.ALPHABET:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SORT.LENGTH:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SORT.DEFAULT:
+    default:
+      break;
+  }
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SORT>(SORT.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const newGoodsFromServer = sort(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': SORT.ALPHABET !== sortField,
+          })}
+          onClick={() => {
+            setSortField(SORT.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': SORT.LENGTH !== sortField,
+          })}
+          onClick={() => {
+            setSortField(SORT.LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(prev => !prev);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || isReversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SORT.DEFAULT);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {newGoodsFromServer.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
When rendering a list, you should provide a unique `key` prop to each child. It's good that you've added a `key` prop to the `li` elements, but using the item's name as a key is risky if the item names are not guaranteed to be unique. It's better to use a unique identifier like an ID.